react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(state => ({
+      isClicked: true
+    }));
```

Comment:
Although you can pass `setState` a callback function, most of the time it is better to just pass it an object with the properties you wish to update. For example:

```JSX
this.setState({
  isClicked: true
});